api/src/main/java/org/pmiops/workbench/api/ConceptSetsController.java

```diff
@@ -354,41 +362,26 @@ private DbConceptSet getDbConceptSet(
         workspaceService.getWorkspaceEnforceAccessLevelAndSetCdrVersion(
             workspaceNamespace, workspaceId, workspaceAccessLevel);
 
-    DbConceptSet conceptSet = conceptSetService.findOne(conceptSetId);
-    if (conceptSet == null || workspace.getWorkspaceId() != conceptSet.getWorkspaceId()) {
-      throw new NotFoundException(
-          String.format(
-              "No concept set with ID %s in workspace %s.",
-              conceptSetId, workspace.getFirecloudName()));
-    }
+    DbConceptSet conceptSet =
+        conceptSetService
+            .findOne(conceptSetId, workspace.getWorkspaceId())
+            .orElseThrow(
+                () ->
+                    new NotFoundException(
+                        String.format(
+                            "No concept set with ID %s in workspace %s.",
+                            conceptSetId, workspace.getFirecloudName())));
```

Comment:
Refactored findOne to take conceptSetId and workspaceId to satisfy both calls and exceptions

Reply:
Cool. FWIW,  you could push this down. into the servcie as well, and have one method that returns an `Optional<ConceptSet>` and another that throws your exception. Since you're doing this twice, that might clean. up. your controller code.